src/mono/wasm/debugger/BrowserDebugProxy/MemberReferenceResolver.cs

```diff
@@ -545,6 +545,54 @@ async Task<ArraySegment<byte>> WriteLiteralExpressionAsIndex(DotnetObjectId root
             }
         }
 
+        private static bool CheckParametersCompatibility(ElementType? typeCode, JObject value)
+        {
+            if (!typeCode.HasValue)
+                return true;
+            var parameterType = value["type"]?.Value<string>();
+            var parameterClassName = value["className"]?.Value<string>();
+
+            switch (typeCode.Value)
+            {
+                case ElementType.Object:
+                    if (parameterType != "object")
+                        return false;
+                    break;
+                case ElementType.I2:
+                case ElementType.I4:
+                case ElementType.I8:
+                case ElementType.R4:
+                case ElementType.R8:
+                case ElementType.U2:
+                case ElementType.U4:
+                case ElementType.U8:
+                    if (parameterType != "number")
+                        return false;
+                    if (parameterType == "object")
+                        return false;
+                    break;
+                case ElementType.Char:
+                    if (parameterType != "string" && parameterType != "symbol")
+                        return false;
+                    break;
+                case ElementType.Boolean:
+                    if (parameterType == "boolean")
+                        return true;
+                    if (parameterType == "number" && (parameterClassName == "Single" || parameterClassName == "Double"))
+                        return false;
+                    if (parameterType == "object")
+                        return false;
+                    break;
+                case ElementType.String:
+                    if (parameterType != "string")
+                        return false;
+                    break;
+                default:
```

Comment:
How do we handle a nullable parameter type?

Reply:
we don't handle it right now.